vassal-app/src/main/java/VASSAL/build/module/chessclockcontrol/ChessClock.java

```diff
@@ -0,0 +1,826 @@
+ * players' computers. A pair (or more, or I suppose only one) of them are then added to {@link ChessClockControl}.
+ *
+ * @author Brian Reynolds and Michael Kiefte
+ *
+ * Class originally created by Michael Kiefte for Twilight Struggle, including handshaking for verification of time totals between online
+ * machines. Made more generically configurable, ChessClockControl added, and integrated into standard VASSAL, by Brian Reynolds.
+ *
+ */
+public class ChessClock extends AbstractConfigurable implements CommandEncoder, GameComponent, ActionListener {
+  private static final int MILLISECONDS_PER_MINUTE = 1000 * 60;
+  private static final int MILLISECONDS_PER_HOUR = MILLISECONDS_PER_MINUTE * 60;
+  private static final int MILLISECONDS_PER_DAY = MILLISECONDS_PER_HOUR * 24;
+  public static final char DELIMITER = '\t';
+  public static final String COMMAND_PREFIX = "CLOCK" + DELIMITER; //$NON-NLS-1$
+
+  public static final String ICON = "icon"; //$NON-NLS-1$
+  public static final String SIDE = "side"; //$NON-NLS-1$
+  public static final String TOOLTIP = "tooltip"; //$NON-NLS-1$
+  public static final String BUTTON_TEXT = "buttonText"; //$NON-NLS-1$
+
+  public static final String TICKING_BACKGROUND_COLOR = "tickingBackgroundColor";
+  public static final String TICKING_FONT_COLOR       = "tickingFontColor";
+  public static final String TOCKING_FONT_COLOR       = "tockingFontColor";
+
+  public static final String GENERIC   = "Player";
+
+  private static Color defaultColor;       // Stores default look-and-feel colors for buttons
+  private static Color defaultFontColor;
+
+  protected Color tickingBackgroundColor = new Color(255, 255, 0);  // Starting (default) colors
+  protected Color tickingFontColor       = new Color(51, 51, 51);
+  protected Color tockingFontColor       = new Color(0, 0, 0);
+
+  protected static Timer timer;    // Millisecond timer (but only actually updated every ~100ms)
+  protected long startTime = -1;   // Millisecond timer count of last timer check
+
+  protected LaunchButton timerButton;  // Button (and display) for this clock
+  protected String image;              // Icon for this clock
+  protected String side;               // Name of side/player whose time this clock represents
+  protected String buttonText;         // Button text for clock button
+  protected long elapsedTime;          // Apparent elapsed time
+  protected long verifiedTime;         // Time successfully verified through handshake
+  protected boolean clockTicking;      // Is this clock running
+
+  protected boolean tocking; // Tracks the half-second "pulse" for the font color
+
+  protected boolean instanceIsActive;  // True if this ChessClock has been fully initialized & registered (and not yet shut down)
+
+
+  public ChessClock(String side) {
+    this();
+    this.side       = side;
+    this.buttonText = side;
+    setConfigureName(side);
+    setTimerButton();
+  }
+
+
+  public ChessClock() {
```

Comment:
Could the `ChessClockControl` be passed in to this ctor instead of being fetched using static methods?